docs/migration-guide.md

```diff
@@ -16,46 +16,57 @@ To ensure AWS your account is set up correctly, run:
 
-Pay attention to the output of the command as it may describe further manual steps at this point.
-Once done, we recommend running `git diff` to see the full set of changes.
+### Introduce CDK to the repository with CI/CD
+1. Create a [new project](setting-up-a-gucdk-project.md) specifying the `--yaml-template-location` flag.
+2. Check to see what changes will be made to the stack:
 
-Finally, confirm that your new CDK stack is not materially different from your existing stack by running a diff:
+   ```shell
+   npm run diff -- --profile <AWS PROFILE NAME> <STACK ID FROM bin/cdk.ts>
+   ```
 
-```sh
-npm run diff
-```
-
-You might find that the diff contains a lot of noise due to tags that are automatically added by `@guardian/cdk`.
-Tag changes will not cause disruption to your infrastructure.
-If desired, you can view a diff which excludes the tagging changes by temporarily [amending your stack's props](https://github.com/guardian/cdk-playground/blob/26af40f6432cb94b3e5ef34648d422a7a6ee6b33/cdk/bin/cdk.ts#L6).
+   This should only show differences in how resources are tagged; GuCDK will add various tags to _all_ resources in the stack.
+   If you find this initial diff too noisy, you could temporarily exclude these tags by amending your [stack's props](https://guardian.github.io/cdk/interfaces/constructs_core.GuStackProps.html#withoutTags),
+   specifically set `withoutTags` to `true`.
 
-Once you are happy with the diff, we recommend running CDK synth as part of your CI process.
-See [here](./setting-up-a-gucdk-project.md) for more detail on CI/CD.
+4. Configure CI and CD. See [here](setting-up-a-gucdk-project.md) for more detail.
+5. Raise a PR and merge.
 
-## Migrating resources
+### Migration to CDK
+We now have something like this:
 
-After initial setup, you have something like this:
+```
+CDK(cfn.yaml) -> cfn.json
+```
 
-    CDK(cfn.yaml) -> cfn.yaml
+That is, CDK is merely wrapping an existing template. The end goal is for CDK to do all the work:
 
-That is, CDK is merely wrapping an existing template. This is useful as a
-starting point. For example, to integrate with CI/CD.
+```
+CDK -> cfn.json
+```
 
-The end goal, though, is for CDK to do all of the work:
+In the previous phase, we also created a [snapshot test](https://docs.aws.amazon.com/cdk/v2/guide/testing.html#testing_snapshot) (`cdk/lib/*.test.ts`),
+which we can treat as the source of truth, representing the exact resources in the stack, and we can now begin the migration process.
 
-    CDK -> cfn.yaml
+As a first step, it is a good idea to read through the template, understanding which resources are [stateful](stateful-resources.md) or not.
+Pro-tip: Annotate the template with inline comments and get it reviewed by your team - it might transpire that a resource is no longer needed as the feature that added it has been removed.
 
-To get there, we need to migrate resources defined in the old cfn.yaml template
-into CDK (Typescript).
+For the stateful resources:
```

Comment:
Thanks for adding these steps, it's very clear!

IMO, the ordering of this guide implies that we want to migrate stateful resources _before_ we migrate the stateless parts (i.e. before we look at the pattern-specific migration guides). Do you think that's the approach we want to take?

Personally I have a few reservations about this because I think you can get most of the benefits of `GuCDK` even if you leave something like a Dynamo table in a CFN file permanently.

I would be tempted to pull the advice re stateless resource migration into its own file and link out to it at the end of the 'pattern' migrations e.g. [here](https://github.com/guardian/cdk/blame/main/docs/migration-guide-ec2.md#L182-L185).

Reply:
How do the latest edits read?